manifests/domain.pp

```diff
@@ -70,60 +70,59 @@
-  unless empty($entry_cache_group_timeout) { validate_integer($entry_cache_group_timeout) }
-  unless empty($entry_cache_netgroup_timeout) { validate_integer($entry_cache_netgroup_timeout) }
-  unless empty($entry_cache_service_timeout) { validate_integer($entry_cache_service_timeout) }
-  unless empty($entry_cache_sudo_timeout) { validate_integer($entry_cache_sudo_timeout) }
-  unless empty($entry_cache_autofs_timeout) { validate_integer($entry_cache_autofs_timeout) }
-  unless empty($entry_cache_ssh_host_timeout) { validate_integer($entry_cache_ssh_host_timeout) }
-  unless empty($refresh_expired_interval) { validate_integer($refresh_expired_interval) }
-  validate_bool($cache_credentials)
-  validate_integer($account_cache_expiration)
-  unless empty($pwd_expiration_warning) { validate_integer($pwd_expiration_warning) }
-  unless empty($use_fully_qualified_names) { validate_bool($use_fully_qualified_names) }
-  unless empty($ignore_group_members) { validate_bool($ignore_group_members) }
-  validate_string($auth_provider)
-  unless empty($auth_provider) { validate_array_member($auth_provider,['ldap','krb5','ipa','ad','proxy','local','none']) }
-  validate_string($access_provider)
-  unless empty($access_provider) { validate_array_member($access_provider,['permit','deny','ldap','ipa','ad','simple']) }
-  validate_string($chpass_provider)
-  unless empty($chpass_provider) { validate_array_member($chpass_provider,['ldap','krb5','ipa','ad','proxy','none']) }
-  validate_string($sudo_provider)
-  unless empty($sudo_provider) { validate_array_member($sudo_provider,['ldap','ipa','ad','none']) }
-  validate_string($selinux_provider)
-  unless empty($selinux_provider) { validate_array_member($selinux_provider,['ipa','none']) }
-  validate_string($subdomains_provider)
-  unless empty($subdomains_provider) { validate_array_member($subdomains_provider,['ipa','ad','none']) }
-  validate_string($autofs_provider)
-  unless empty($autofs_provider) { validate_array_member($autofs_provider,['ldap','ipa','none']) }
-  validate_string($hostid_provider)
-  unless empty($hostid_provider) { validate_array_member($hostid_provider,['ipa','none']) }
-  validate_string($re_expression)
-  validate_string($full_name_format)
-  validate_string($lookup_family_order)
-  unless empty($lookup_family_order) { validate_array_member($lookup_family_order,['ipv4_first','ipv4_only','ipv6_first','ipv6_only']) }
-  validate_integer($dns_resolver_timeout)
-  validate_string($dns_discovery_domain)
-  validate_string($override_gid)
-  validate_string($case_sensitive)
-  unless empty($case_sensitive) { validate_array_member($case_sensitive,['true','false','preserving']) }
-  unless empty($proxy_fast_alias) { validate_bool($proxy_fast_alias) }
-  validate_string($realmd_tags)
-  validate_string($proxy_pam_target)
-  validate_string($proxy_lib_name)
+#  validate_array_member($id_provider,['proxy','local','ldap','ipa','ad'])
```

Comment:
I am saving the validations because we decided that was how were are going to do it for now and they have some data in them that can be used when we strong type all the variables.